### ISSUE TITLE
Is there a way of implementing the eventComparator without using getTime?

### ISSUE THREAD

**Issue by yuqitanyq** [nus-cs2030/2122-s1#80]
# Summary
I was just wondering if there is any way of implementing the eventComparator for the priotiy queue without having the need to use getTime?

## Description 
I can't really think of a way to not use getTime and still use time as a basis of comparison? Also, if the timings are the same, what would the tiebreaker be based on? Would appreciate any help!

**Comment by EricBryann** [comment]
Not sure how not to use `getTime` but I used `getTime`!
My tiebraker is that the smaller customer Id will have higher priority

**Comment by nmtuan2001** [comment]
I don't think there is any way to implement the eventComparator other than adding a getter for time. For the tiebreaker, I think it should be time, then customerId (smaller id first), and finally event priority (for example, arrive < wait < serve < done < leave).

**Comment by BrendanCheong** [comment]
> I don't think there is any way to implement the eventComparator other than adding a getter for time. For the tiebreaker, I think it should be time, then customerId (smaller id first), and finally event priority (for example, arrive < wait < serve < done < leave).

Agreed. I did the same for the tiebreaker

**Comment by jeromepui** [comment]
> I don't think there is any way to implement the eventComparator other than adding a getter for time. For the tiebreaker, I think it should be time, then customerId (smaller id first), and finally event priority (for example, arrive < wait < serve < done < leave).

Yup, I assigned numbers to the different events (1,2,3,4,5) to prioritise them, and then compared the time using getTime() in the case of a tiebreaker.

**Comment by Le-PenDu** [comment]
You could implement the Comparable interface in your Event class, and make the comparisons there instead

**Comment by JustinS00** [comment]
Yup, following up on the point above, i implemented Comparable<Event> and Comparable<Customer> for the two classes respectively, for Event.compareTo(Event other) i compared the this.time and other.time first, if its the same then i will call for this.customer.compareTo(other.customer)

**Comment by chenghong99** [comment]
I used a getTime(), if do not know if there is any other way to go about it without using a getTime(), since the priority is based on time.
